### PR TITLE
Nanotrasen Deserter Vault Ruin

### PR DESCRIPTION
**Pull request by Latentish**
## About The Pull Request
Adds a new ruin spawn to the icemoon called the "Nanotrasen Corporate Security Solutions Vault Gamma". The level is full of Nanotrasen deserters guarding a very secure vault. The level takes into account players who might want to plan a heist rather than run through the level in a normal manner, beefing up defenses in areas one might try to break in through. It would take about the same amount of effort to plan and mount a heist on the vault as it would to run through it plain and simple. The level is possible to complete with no pre-brought medical supplies, running as the Hyena captain alone, so it should be fine for other players to complete as well. The total pool of loot that can be gotten is as follows:
Weapons and Armor: Particle Acceleration Rifle, NT Saber SMG with firing pin, combat shotgun, fireaxe, medibeam, disablers, Runite Armor (platemail armor), 2 spare clips for the smg, upgraded weapon cell for the particle acceleration rifle, and a variety total of 16 loose shotgun shells.
Materials: 10 Capitalisium, 10 runite, 2 adamantine, 15 bananium, 20 gold, 5 bluespace polycrystals, 20 plasma, 20 uranium, 2000 credits.
Machines/vendors: Empty AI core, soda dispenser, booze dispenser, boozeomat, misc civilian vendors, sectech, autolathe.
Equipment: Several basic toolsets, filled surgery duffelbag, stethoscope, medkit, rolling table, welding skull gasmask, insulated gloves, CNS rebooter.
Misc: Security weapon voucher, red phone, top secret documents (folder), hand tele, chaos donut, Oingo Boingo punch-face, positronic brain, MMI, HONK central control module, posters, bedsheets, sentience potion.

image
image
image
image

## Why It's Good For The Game
The Icemoon is rather lacking in number of ruins when compared to other planets. This level adds some higher level loot to the game, while fitting the niche for a more difficult challenge. 

## Changelog
:cl:Phoaly
add: Added a new map "maps\randomruins\iceruins\icemoon_surface_corporate_rejects.dmm"
code: Added a datum and several areas for the map in "whitesands\code\datums\ruins\icemoon.dm" and "code\game\area\areas\ruins\icemoon.dm" respectively.
/:cl:

### CODE REVIEW

**Comment by Latentish** [comment]
Feedback on the loot pool would be appreciated.

**Comment by UltraNumeron** [comment]
Seems pretty good honestly only issue i have is that for the loot pool it seems a bit too easy if im being honest, could maybe add some turrets on the outside if you can get those to work with maybe some enemies inside the vault itself

**Comment by Firefox13** [comment]
> Feedback on the loot pool would be appreciated.

R&D documents

**Comment by Latentish** [comment]
> 

I don't really understand what that means.

**Comment by UltraNumeron** [comment]
> > 
> 
> I don't really understand what that means.

They likely mean papers containing rnd points, like the discoveries you can get from  dissecting corpses and then slap in a research console for points.

**Comment by Latentish** [comment]
> > > 
> > 
> > 
> > I don't really understand what that means.
> 
> They likely mean papers containing rnd points, like the discoveries you can get from dissecting corpses and then slap in a research console for points.

Makes sense, I'll swap those out for something else.

**Comment by 1glitchycent** [comment]
replace the sniper rifle with a tactical or antique Egun, NT doesn't into ballistics

**Comment by Latentish** [comment]
> replace the sniper rifle with a tactical or antique Egun, NT doesn't into ballistics

I mean, they have their own brand of SMG. But people were having an issue with the sniper rifle anyway, so I'll find another funny gun. Maybe a pulse rifle?

**Comment by Apogee-dev** [comment]
NT definitely does into ballistics, but the sniper rifle isn't one of theirs and is seriously busted. Make it a captain's laser gun instead, that's high quality loot without being outright broken.

**Comment by Latentish** [comment]
What are run linters?

File: code/game/area/areas/ruins/icemoon.dm [hunk 1]
```diff
@@ -38,3 +38,16 @@
...
+
+/area/ruin/unpowered/corprejectrooms
+	name = "NT Security Solutions Site Gamma Rooms"
+	icon_state = "dk_yellow"
+
+/area/ruin/unpowered/corprejectengineering
+	name = "NT Security Solutions Site Gamma Engineering"
+	icon_state = "dk_yellow"
+
+	/area/ruin/unpowered/corprejectvault
```
Comment (ZephyrTFA) [inline code/game/area/areas/ruins/icemoon.dm]:
```suggestion
/area/ruin/unpowered/corprejectvault
```

**Comment by Kapu1178** [comment]
> What are run linters?

linters are the per-compilation scan of the code to find things that are wrong with it.

**Comment by Latentish** [comment]
Mergema and/or reviewma.

File: whitesands/code/datums/ruins/icemoon.dm [hunk 2]
```diff
@@ -34,3 +34,9 @@
 	id = "demonlab"
 	description = "Some abandoned research outpost"
 	suffix = "icemoon_demonlab.dmm"
+
+/datum/map_template/ruin/icemoon/corpreject
+	name = "NT Security Solutions Site Gamma"
+	id = "corpreject"
+	description = "Nanotrasen Corporate Security Solutions vault site Gamma."
+	suffix = "icemoon_surface_corporate_rejects.dmm"
```
Comment (triplezeta) [inline whitesands/code/datums/ruins/icemoon.dm]: Move to the unmodular ruin datums file please

**Review by triplezeta — CHANGES_REQUESTED** [review]
Solid looking ruin overall, just need to address this modularization bit

**Comment by triplezeta** [comment]
Balance issues with the ruin can be addressed after a few rounds of live tests

File: whitesands/code/datums/ruins/icemoon.dm (hunk 2 above)
Comment (Latentish) [reply whitesands/code/datums/ruins/icemoon.dm]: Donezo!

**Review by ZephyrTFA — APPROVED** [review]
I don't really approve maps, but we also don't have a dedicated maptainer so I'll toss my approval on this

**Comment by Latentish** [comment]
> I don't really approve maps, but we also don't have a dedicated maptainer so I'll toss my approval on this

Glad you like it at the least! =)(

**Pull request merged**